ts/Stock/Indicators/DPOIndicator.ts

```diff
@@ -1,217 +0,0 @@
- * @private
- */
-declare global {
-    namespace Highcharts {
-        class DPOIndicator extends SMAIndicator {
-            public data: Array<DPOIndicatorPoint>;
-            public nameBase: string;
-            public options: DPOIndicatorOptions;
-            public pointClass: typeof DPOIndicatorPoint;
-            public points: Array<DPOIndicatorPoint>;
-            public getValues<TLinkedSeries extends LineSeries>(
-                series: TLinkedSeries,
-                params: DPOIndicatorParamsOptions
-            ): (IndicatorValuesObject<TLinkedSeries>|undefined);
-        }
-
-        interface DPOIndicatorOptions extends SMAOptions {
-            params?: DPOIndicatorParamsOptions;
-        }
-
-        interface DPOIndicatorParamsOptions extends SMAParamsOptions {
-            // for inheritance
-        }
-
-        class DPOIndicatorPoint extends SMAPoint {
-            public series: DPOIndicator;
-        }
-    }
-}
-
-declare module '../../Core/Series/SeriesType' {
-    interface SeriesTypeRegistry {
-        dpo: typeof Highcharts.DPOIndicator;
-    }
-}
-
-// im port './SMAIndicator.js';
-
-/* eslint-disable valid-jsdoc */
-// Utils
-/**
- * @private
- */
-function accumulatePoints(
-    sum: number,
-    yVal: (Array<number>|Array<Array<number>>),
-    i: number,
-    index: number,
-    subtract?: boolean
-): number {
-    var price = pick<(number|undefined), number>(
-        (yVal[i] as any)[index], (yVal[i] as any)
-    );
-
-    if (subtract) {
-        return correctFloat(sum - price);
-    }
-    return correctFloat(sum + price);
-}
-
-/* eslint-enable valid-jsdoc */
-
-/**
- * The DPO series type.
- *
- * @private
- * @class
- * @name Highcharts.seriesTypes.dpo
- *
- * @augments Highcharts.Series
- */
-BaseSeries.seriesType<typeof Highcharts.DPOIndicator>(
-    'dpo',
-    'sma',
-    /**
-     * Detrended Price Oscillator. This series requires the `linkedTo` option to
-     * be set and should be loaded after the `stock/indicators/indicators.js`.
-     *
-     * @sample {highstock} stock/indicators/dpo
-     *         Detrended Price Oscillator
-     *
-     * @extends      plotOptions.sma
-     * @since        7.0.0
-     * @product      highstock
-     * @excluding    allAreas, colorAxis, compare, compareBase, joinBy, keys,
-     *               navigatorOptions, pointInterval, pointIntervalUnit,
-     *               pointPlacement, pointRange, pointStart, showInNavigator,
-     *               stacking
-     * @requires     stock/indicators/indicators
-     * @requires     stock/indicators/dpo
-     * @optionparent plotOptions.dpo
-     */
-    {
-        /**
-         * Parameters used in calculation of Detrended Price Oscillator series
-         * points.
-         */
-        params: {
-            /**
-             * Period for Detrended Price Oscillator
-             */
-            period: 21
-        }
-    },
-    /**
-     * @lends Highcharts.Series#
-     */
-    {
-        nameBase: 'DPO',
```

Comment:
I believe that the value of nameBase somehow got removed. This part needs to be restored with the `extend` call right under the interface of the prototype properties.
```ts
extend(DPOIndicator.prototype, {
    nameBase: 'DPO'
});
```